File: 06.py
#!/usr/bin/env python3
from sys import stdin
from functools import reduce

init  = list(map(int, stdin.read().strip().split(',')))
arr   = [0,0,0,0,0,0,0,0,0]
for x in init: arr[x] += 1
tick  = lambda a,b,c,d,e,f,g,h,i: [b,c,d,e,f,g,a+h,i,a]
ticks = lambda m: sum (reduce (lambda a,n: tick(*a), range(m), arr))

print(f"Part 1: { ticks(80) }, Part 2: { ticks(256) }")

File: 06.scala
#!/usr/bin/env scala
import scala.io.Source.stdin

@main def aoc(): Unit =
  val init = stdin.getLines().flatMap(_.split(',')).map(_.toInt)
            .foldLeft(Vector.fill(9)(0L))((a, i)=>a.updated(i, a(i)+1)).toList
  val tick = (x:List[Long]) => List(x(1),x(2),x(3),x(4),x(5),x(6),x(0)+x(7),x(8),x(0))
  val ticks = (n:Int) => (1 to n).foldLeft(init)((a,_)=>tick(a)).sum

  println(s"Part 1: ${ ticks(80) }, Part 2: ${ ticks(256) }")
